### PR TITLE
feat(sqlsmith): updates and deletes

### LINKED ISSUE #9824: Sqlsmith: updates and deletes
- [x] Modify sqlsmith table struct to support PK, so update can be on datum.
- [x] Store inserted keys, we can sample these to update.
- [x] Generate updates and deletes AFTER mview step.

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

 Around ~500 lines are from the snapshot tests.

Sqlsmith currently does the following:
1. Generates base tables.
2. Generate inserts
3. Generates mviews on top of these base tables.
4. Test batch + stream query.

After this PR:
1. Generates base tables.
2. Generate inserts
3. Generates mviews on top of these base tables.
4. **+++ Generate updates (if have pk) / delete + inserts (if no pk)**
5. Test batch + stream query.

I found that interleaving inserts, updates and deletes between stream queries is complicated to debug. I have left it out for now. It can be done in the future.

Updates and deletes are generated by:
1. Finding PK of tables.
1. Selecting a subset of inserts.
1. Selecting their PK, and generating new values for these rows.
1. If they do not have PK, just delete some rows, generate some new inserts.

Closes #9824.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/tests/sqlsmith/src/sql_gen/insert.rs [hunk 1]
```diff
@@ -1,59 +0,0 @@
-// Copyright 2023 RisingWave Labs
```
Comment (kwannoel) [inline src/tests/sqlsmith/src/sql_gen/insert.rs:1]: Moved to `dml.rs`

File: src/tests/sqlsmith/src/sql_gen/dml.rs [hunk 2]
```diff
@@ -0,0 +1,233 @@
+// Copyright 2023 RisingWave Labs
```
Comment (kwannoel) [inline src/tests/sqlsmith/src/sql_gen/dml.rs:1]: This contains the update and delete generation logic.

File: src/tests/sqlsmith/src/runner.rs [hunk 3]
```diff
@@ -86,12 +88,16 @@ pub async fn generate(
...
         tables.clone(),
-        base_tables,
+        base_tables.clone(),
         max_rows_inserted,
     )
     .await;
     tracing::info!("Passed sqlsmith tests");
 
+    // Generate an update for some inserts, on the corresponding table.
+    update_base_tables(client, &mut rng, &base_tables, &inserts).await;
```
Comment (fuyufjh) [inline src/tests/sqlsmith/src/runner.rs:98]: Why `update_base_tables` is after `test_sqlsmith`?

**Review by fuyufjh — APPROVED** [review]

**Comment by kwannoel** [comment]
Commits are solely from rebasing.

**Pull request merged**